duecredit/collector.py

```diff
@@ -141,7 +149,7 @@ def objname(self):
         else:
             return None
 
-    def __contains__(self, entry):
+    def __contains__(self, entry: 'Citation') -> bool:  # Self PEP673
```

Comment:
If, as the comment suggests, `entry` is only supposed to be of the exact same type as `self` (not even allowing for subclassing), then it should be annotated with `Self` instead of `"Citation"`.  As `Self` is only used as an annotation, and if you take my advice above about `from __future__ import annotations`, you can import `Self` by simply adding this to the top of the file:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing_extensions import Self
```

`typing_extensions` is a dependency of mypy, so you won't even have to adjust the project's dependencies!